doc/design/zarr-support-3.md

```diff
@@ -0,0 +1,146 @@
+# Zarr Support
+A zarr archive contains directories and zarr files. 
+Zarr archives are stored in their entirety in S3.
+Zarr archives are represented by a single Asset through the API.
+A tree hash is computed as the zarr archive is uploaded.
+Each node in the tree hash in the hash is stored in S3 to avoid bloating the DB.
+The intended use of Zarr archives is primarily to support use cases where data are too big to be copied. However, Zarr archives can be small as well.
+
+# Requirements
+1. Zarr archives are stored in a "directory" in S3.
+1. Each zarr archive corresponds to a single Asset.
+1. The CLI uses some kind of tree hashing scheme to compute a checksum for the entire zarr archive.
+1. The API verifies the checksum _immediately_ after upload.
```

Comment:
```suggestion
1. The API verifies the checksum immediately after upload, so the validation status is available in the response to the upload REST request.
```
The alternative would be validating asynchronously, which I was told was not an option. The scheme described here requires a request to S3 for each directory in the paths being updated (`a/b/c/d/e/f` = 5 requests), which should be nowhere near the 30s Heroku request timeout.

Reply:
Is that implemented so I could check the code? I am still a bit lost since upload is a batch of files which could come for various directories, so it could in principle (e.g. a file per directory) require  traversal of even more "directories" (up the hierarchy) than files.